c/include/libsbp/common.h

```diff
@@ -22,6 +22,16 @@
 #ifndef COMMON_INT_TYPES
 #define COMMON_INT_TYPES
 
+/* Determine Toolchain */
+
```

Comment:
This is what I was thinking of:

```c
#if defined(_MSC_VER)

#define SBP_PACK_START \
    __pragma(pack(1));
#define SBP_PACK_END \
    __pragma(pack());

#define SBP_PACKED /* Intentionally empty */

#elif defined(__GNUC__) || defined(__clang__)

#define SBP_PACK_START
#define SBP_PACK_END

#define SBP_PACKED __attribute__((packed))

#else

#if !defined(SBP_PACK_START)
#define SBP_PACK_START
#warning Unknown compiler, please override SBP_PACK_START
#endif

#if !defined(SBP_PACK_END)
#define SBP_PACK_END
#warning Unknown compiler, please override SBP_PACK_END
#endif

#if !defined(SBP_PACKED)
#define SBP_PACKED __attribute__((packed))
#warning Unknown compiler, please override SBP_PACKED
#endif

#endif
```

Reply:
So SBP_PACK_START and SBP_PACK_END would make this look a little cleaner:

```c
#ifdef TOOLCHAIN_PRAGMA_PACK
#pragma pack(1) // or #pragma pack() for the end of the file
#endif
```

Also, using `SBP_PACKED` so it's a little shorter (hopefully less aesthetically offensive) and added the `SBP_` prefix to make it clear it's a macro we're defining/using and will prevent it from colliding with other names (however unlikely that may be) ... we could even make it lowercase `__sbp_attribute_packed__` to make it look more like `__attribute__((packed))`, if we're so inclined.  Anyway, regardless of aesthetics, if we can make it more explicit what compilers we're supporting with the above approach, then I'm fine with this.